tests/framework-tests/core/js/DataSourceTests.js

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2014 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+// Declare dependencies
+/* global fluid, jqUnit */
+
+(function () {
+    "use strict";
+
+    fluid.registerNamespace("fluid.tests");
+
+    jqUnit.test("fluid.toHashKey", function () {
+        var singleLevel = {
+            a: "A",
+            b: "B",
+            g: 5,
+            c: ["D", "E", "F"]
+        };
+
+        var nested = {
+            a: "A",
+            h: 6,
+            f: ["F", "G"],
+            b: {
+                c: "C",
+                d: ["D", "E"]
+            }
+        };
+
+        var expected = {
+            singleLevel: "<object a:|\"A\"|b:|\"B\"|c:<array 0:|\"D\"|1:|\"E\"|2:|\"F\"|>g:|5|>",
+            nested: "<object a:|\"A\"|b:<object c:|\"C\"|d:<array 0:|\"D\"|1:|\"E\"|>>f:<array 0:|\"F\"|1:|\"G\"|>h:|6|>"
+        };
+
+        jqUnit.assertEquals("The single level object should be converted properly", expected.singleLevel, fluid.toHashKey(singleLevel));
```

Comment:
I've taken your example and done an assertEquals between the outputs of calling fluid.toHashKey on order1 and order2.